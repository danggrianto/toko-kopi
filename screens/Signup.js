import React, { useState } from 'react'
import { TextInput, Button, View, Text, StyleSheet } from 'react-native'

import { firebase } from '../config/Firebase'
import Colors from '../config/Colors'

function Signup({ navigation }) {
  const [name, setName] = useState('')
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [confirmPassword, setConfirmPassword] = useState('')

  const onRegisterPress = () => {
    if (password !== confirmPassword) {
      alert("Passwords don't match.")
      return
    }
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then((response) => {
        const uid = response.user.uid
        const data = {
          id: uid,
          email,
          name,
        }
        const usersRef = firebase.firestore().collection('users')
        usersRef
          .doc(uid)
          .set(data)
          .then(() => {
            navigation.navigate('My Account', { user: data })
          })
          .catch((error) => {
            alert(error)
          })
      })
      .catch((error) => {
        alert(error)
      })
  }

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.inputBox}
        placeholder="Name"
        placeholderTextColor="grey"
        onChangeText={(text) => setName(text)}
        value={name}
      />
      <TextInput
        style={styles.inputBox}
        placeholder="Email"
        placeholderTextColor="grey"
        autoCapitalize="none"
        onChangeText={(text) => setEmail(text)}
        value={email}
      />
      <TextInput
        style={styles.inputBox}
        value={password}
        onChangeText={(text) => setPassword(text)}
        placeholder="Password"
        placeholderTextColor="grey"
        secureTextEntry={true}
      />
      <TextInput
        style={styles.inputBox}
        value={confirmPassword}
        onChangeText={(text) => setConfirmPassword(text)}
        placeholder="Confirm Password"
        placeholderTextColor="grey"
        secureTextEntry={true}
      />
      <Button
        title="Sign me up!"
        color={Colors.primary}
        onPress={onRegisterPress}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  inputBox: {
    width: '85%',
    margin: 10,
    padding: 15,
    fontSize: 16,
    borderColor: Colors.outline,
    borderBottomWidth: 1,
    color: Colors.outline,
    fontWeight: 'bold',
    textAlign: 'center',
  },
})

export default Signup
