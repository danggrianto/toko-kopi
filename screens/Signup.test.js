import React from 'react'
import renderer from 'react-test-renderer'
import Signup from './Signup'

test('renders correctly', () => {
  const tree = renderer.create(<Signup />).toJSON()
  expect(tree).toMatchSnapshot()
})
