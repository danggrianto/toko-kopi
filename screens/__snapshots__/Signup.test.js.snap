// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Name"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#6D4539",
        "color": "#6D4539",
        "fontSize": 16,
        "fontWeight": "bold",
        "margin": 10,
        "padding": 15,
        "textAlign": "center",
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#6D4539",
        "color": "#6D4539",
        "fontSize": 16,
        "fontWeight": "bold",
        "margin": 10,
        "padding": 15,
        "textAlign": "center",
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#6D4539",
        "color": "#6D4539",
        "fontSize": 16,
        "fontWeight": "bold",
        "margin": 10,
        "padding": 15,
        "textAlign": "center",
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Confirm Password"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#6D4539",
        "color": "#6D4539",
        "fontSize": 16,
        "fontWeight": "bold",
        "margin": 10,
        "padding": 15,
        "textAlign": "center",
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#F8AF5E",
            },
          ]
        }
      >
        Sign me up!
      </Text>
    </View>
  </View>
</View>
`;
